### ISSUE TITLE
Resize helper for plot functions with a view

### ISSUE THREAD

**Issue by zachbogart** [observablehq/framework#1211]
The [resize helper example](https://observablehq.com/framework/javascript#responsive-display) works for a Plot function that accepts a width and/or height. The helper should allow for the ability to resize _and_ have access to a view, such as a tooltip mark where you can access the data on hover.

- See [chart as input example](https://observablehq.com/framework/reactivity#inputs) from docs. Want to allow for that functionality, but also allow for resizing.

**Comment by mbostock** [comment]
Yep, I have a workaround in https://talk.observablehq.com/t/framework-using-resize-with-view/8895/3:

```js
function valueof(element, initialValue = null) {
  return Generators.observe((change) => {
    const input = (event) => change(event.target.value);
    change(initialValue);
    element.addEventListener("input", input);
    return () => element.removeEventListener("input", input);
  });
}
```

Then use `valueof` in place of `Generators.input`:

```js
const penguinChart = resize((width) => Plot.dot(penguins, {x: "culmen_length_mm", y: "culmen_depth_mm", tip: true}).plot({width}));
const penguin = valueof(penguinChart);
```

We should probably just change `Generators.input` (and by extension `view`) to work this way by default, possibly with a few more tweaks.

**Comment by zachbogart** [comment]
Cool, that gets the job done. Agreed would be nice for the other view helpers to work that way by default.
